tsdb/index/tsi1/partition.go

```diff
@@ -909,6 +952,30 @@ func (p *Partition) compact() {
 	fs := p.retainFileSet()
 	defer fs.Release()
 
+	// compact any non-active log files first
+	for _, f := range p.fileSet.files {
+		if f.Level() == 0 {
+			logFile := f.(*LogFile) // It is an invariant that a file is level 0 iff it is a log file
+			if logFile == p.activeLogFile {
+				continue
+			}
+			if p.levelCompacting[0] {
+				break
+			}
+			// Mark the level as compacting.
+			p.levelCompacting[0] = true
+			p.currentCompactionN++
+			go func() {
+				p.compactLogFile(logFile)
```

Comment:
Maybe an error return from `compactLogFile` that aborts the subsequent compaction?

Reply:
We would unconditionally try to call compact after a log file compaction before, so I think this is ok.